Assets/MRTK/Providers/WindowsMixedReality/Shared/InputHandlers/HPMotionControllerInputHandler.cs

```diff
@@ -0,0 +1,264 @@
+                for (int i = 0; i < Interactions?.Length; i++)
+                {
+                    switch (Interactions[i].AxisType)
+                    {
+                        case AxisType.None:
+                            break;
+                        case AxisType.Digital:
+                            UpdateButtonData(Interactions[i], controllerState);
+                            break;
+                        case AxisType.SingleAxis:
+                            UpdateSingleAxisData(Interactions[i], controllerState);
+                            break;
+                        case AxisType.DualAxis:
+                            UpdateDualAxisData(Interactions[i], controllerState);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateButtonDataPerfMarker = new ProfilerMarker("[MRTK] HPMotionControllerInputHander.UpdateButtonData");
+
+        /// <summary>
+        /// Update an interaction bool data type from a bool input
+        /// </summary>
+        /// <remarks>
+        /// Raises an Input System "Input Down" event when the key is down, and raises an "Input Up" when it is released (e.g. a Button)
+        /// </remarks>
+        internal virtual void UpdateButtonData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateButtonDataPerfMarker.Auto())
+            {
+                // Handedness must be left or right in order to differentiate between buttons for the left and right hand.
+                MotionControllerHandedness controllerHandedness = controllerState.MotionController.Handedness;
+
+                Debug.Assert(controllerHandedness != MotionControllerHandedness.Unknown);
+                Debug.Assert(interactionMapping.AxisType == AxisType.Digital);
+
+                if (interactionMapping.InputType == DeviceInputType.TriggerPress)
+                {
+                    var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                    interactionMapping.BoolData = triggerData.Equals(1);
+                }
+                else if (interactionMapping.InputType == DeviceInputType.GripPress)
+                {
+                    var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                    interactionMapping.BoolData = gripData.Equals(1);
+                }
+                else
+                {
+                    ControllerInput button;
+
+                    // Update the interaction data source
+                    switch (interactionMapping.InputType)
+                    {
+                        case DeviceInputType.Select:
+                        case DeviceInputType.TriggerTouch:
+                        case DeviceInputType.TriggerNearTouch:
```

Comment:
For now, we might want to scope this down to `DeviceInputType`s that are defined in `HPMotionControllerDefinition`. That way it's a little more concise and we don't have potentially undefined behavior if this class is reused.